forge-ai/src/main/java/forge/ai/AiController.java

```diff
@@ -1547,8 +1574,12 @@ private SpellAbility getSpellAbilityToPlay() {
 
         Iterables.removeIf(saList, spellAbility -> { //don't include removedAI cards if somehow the AI can play the ability or gain control of unsupported card
             // TODO allow when experimental profile?
-            return spellAbility.isLandAbility() || (spellAbility.getHostCard() != null && ComputerUtilCard.isCardRemAIDeck(spellAbility.getHostCard()));
+            return spellAbility.isLandAbility() || (spellAbility.getHostCard() != null && ComputerUtilCard.isCardRemAIDeck(spellAbility.getHostCard()))  || !spellAbility.canCastTiming(player);
```

Comment:
does this fix anything? it's already checked by canPlaySa so redundant check just adds extra load

Reply:
on my test, the amount of sa is reduced though the canplaysa check is used for counter, the saList = ComputerUtilAbility.getSpellAbilities(cards, player); don't have any filters for cancasttiming.

Reply:
I don't understand what you're saying but pasting checks left and right like that is just asking for trouble
The SA isn't fully prepared for checking at this point, you will produce some misplay